### ISSUE TITLE
Limnoria requests unsupported capabilities instead of watching CAP LS

### ISSUE THREAD

**Issue by Mikaela** [ProgVal/Limnoria#1115]
```
2015-05-20 23:06:59+0300 < Mikaela> are clients supposed to request unsupported capabilities from server? Limnoria released earlier today complains about freenode refusing metadata-notify and account-tag and I think they don't appear in CAP LS (or was it CAP LIST)
2015-05-20 23:07:18+0300 < SaberUK> Mikaela: no
2015-05-20 23:07:27+0300 < Mikaela> pinkieval: ^^
2015-05-20 23:07:30+0300 < SaberUK> Mikaela: you're supposed to cap ls and then see what the server supports
2015-05-20 23:07:45+0300 < Mikaela> I think so too, but Limnoria is apparently misbehaving
2015-05-20 23:08:02+0300 < Mikaela> 2015-05-20 23:07:53+0300 <@Dysforia> The current (running) version of this Supybot is 0.83.4.1+limnoria 2015.05.20, running on Python 3.4.3 (default, May  9 2015, 06:38:10)  [GCC 4.7.2].  The newest versions available online are 2015.05.20 (in master), 2015.05.20 (in testing).
2015-05-20 23:09:16+0300 < Mikaela> I should probably open an issue instead of talking about it here...
```

**Comment by ProgVal** [comment]
As discussed later on the channel, nothing in the spec says it's mandatory.

**Comment by Mikaela** [comment]
But I believe it's stupid behaviour to first ask what the server supports and then spam the server with capabilities it doesn't support.

**Comment by DarthGandalf** [comment]
If response of `CAP LS` is ignored, no need to send `CAP LS` at all, blind `CAP REQ`/`CAP REQ`/`CAP REQ`/`CAP END` should work.
The described behavior is "stupid", but it doesn't contradict the spec.

Disclaimer: I didn't look how Limnoria does it, this response is based solely on conversation pasted in the first post, and on specs.

**Comment by Mikaela** [comment]
Log from bot starting up until just before joining channels.

```
INFO 2015-05-24T11:40:02 supybot Completed daemonization.  Current PID: 10215
INFO 2015-05-24T11:40:03 supybot Starting HTTP server: server at 0.0.0.0 61825
INFO 2015-05-24T11:40:03 supybot Starting HTTP server: server at 2001:67c:1be8::2:0:ea 61825
INFO 2015-05-24T11:40:03 supybot Creating new Irc for pirateirc.
INFO 2015-05-24T11:40:03 supybot Connecting to irc.pirateirc.net:6697.
INFO 2015-05-24T11:40:03 supybot Loading plugins (connecting to pirateirc).
INFO 2015-05-24T11:40:05 supybot Creating new Irc for freenode.
INFO 2015-05-24T11:40:05 supybot Connecting to chat.freenode.net:6697.
INFO 2015-05-24T11:40:05 supybot Loading plugins (connecting to freenode).
INFO 2015-05-24T11:40:05 supybot Server stockholm-se.pirateirc.net has version charybdis-3.5.0-dev
INFO 2015-05-24T11:40:06 supybot Got end of MOTD from stockholm-se.pirateirc.net
INFO 2015-05-24T11:40:06 supybot Sending user modes to pirateirc: -i+wRQx
INFO 2015-05-24T11:40:06 supybot pirateirc: Server acknowledged capability 'account-notify'
INFO 2015-05-24T11:40:06 supybot pirateirc: Server acknowledged capability 'extended-join'
INFO 2015-05-24T11:40:06 supybot pirateirc: Server acknowledged capability 'multi-prefix'
WARNING 2015-05-24T11:40:06 supybot pirateirc: Server refused capability 'metadata-notify'
WARNING 2015-05-24T11:40:06 supybot pirateirc: Server refused capability 'account-tag'
INFO 2015-05-24T11:40:06 supybot pirateirc: Server acknowledged capability 'sasl'
ERROR 2015-05-24T11:40:06 supybot Unhandled error message from server: IrcMsg(prefix="stockholm-se.pirateirc.net", command="462", args=('Dysforia', 'You may not reregister'))
INFO 2015-05-24T11:40:06 supybot Server rajaniemi.freenode.net has version ircd-seven-1.1.3
INFO 2015-05-24T11:40:06 supybot Got end of MOTD from rajaniemi.freenode.net
INFO 2015-05-24T11:40:06 supybot Sending user modes to freenode: -i+wRQ
INFO 2015-05-24T11:40:09 supybot Join to #uutiset on pirateirc synced in 3.34 seconds.
INFO 2015-05-24T11:40:12 supybot freenode: Server acknowledged capability 'account-notify'
INFO 2015-05-24T11:40:12 supybot freenode: Server acknowledged capability 'extended-join'
INFO 2015-05-24T11:40:12 supybot freenode: Server acknowledged capability 'multi-prefix'
WARNING 2015-05-24T11:40:12 supybot freenode: Server refused capability 'metadata-notify'
WARNING 2015-05-24T11:40:12 supybot freenode: Server refused capability 'account-tag'
INFO 2015-05-24T11:40:12 supybot freenode: Server acknowledged capability 'sasl'
ERROR 2015-05-24T11:40:13 supybot Unhandled error message from server: IrcMsg(prefix="rajaniemi.freenode.net", command="462", args=('Dysforia', 'You may not reregister'))
```

I am not sure if Limnoria requests `CAP LS` as it's not visible with INFO logging, but I guess it does as the specs require at least that.

**Comment by ProgVal** [comment]
> I am not sure if Limnoria requests `CAP LS` as it's not visible with INFO logging

It does not

> but I guess it does as the specs require at least that.

Can you quote the piece that requires that?

**Comment by ProgVal** [comment]
As someone highlighted it on IRC, it's not mandatory, but it's the only way to tell the server we support IRCv3.2.

Implemented in e2b3b917e5227981979a0b7a65afb88b231eaffc.
